speech_to_text.py

```diff
@@ -20,75 +20,24 @@
 from honeybadger import honeybadger
 from whisper.utils import get_writer
 from mypy_boto3_s3.service_resource import Bucket, S3ServiceResource
-from mypy_boto3_sqs.service_resource import SQSServiceResource, Queue
+from mypy_boto3_sqs.service_resource import Queue
 
-honeybadger.configure(
-    api_key=os.environ.get("HONEYBADGER_API_KEY", ""),
-    environment=os.environ.get("HONEYBADGER_ENV", "stage"),
-)
 
-
-def main(daemon=True) -> None:
-    # loop forever looking for jobs unless daemon option says not to
-    while True:
-        try:
-            job: Optional[Dict] = get_job()
-            if job is None:
-                logging.info("no jobs waiting in the todo queue")
-            else:
-                logging.info(f"starting job {job}")
-                job = download_media(job)
-                job = run_whisper(job)
-                job = upload_results(job)
-                job = finish_job(job)
-                logging.info(f"finished job {job}")
-        except KeyboardInterrupt:
-            logging.info("exiting")
-            sys.exit()
-        except SpeechToTextException as e:
-            report_error(
-                f"Caught error while processing job: {e}", job, traceback.format_exc()
-            )
-        except Exception as e:
-            report_error(f"Caught unexpected error: {e}", job, traceback.format_exc())
-
-        if not daemon:
-            break
-
-
-def get_job() -> Optional[dict]:
-    """
-    Fetch the next job that is queued for processing. If no job is found in WaitTimeSeconds
-    seconds None will be returned.
-    """
-    queue = get_todo_queue()
-    logging.info("looking for messages in the todo queue")
-    jobs = queue.receive_messages(MaxNumberOfMessages=1, WaitTimeSeconds=20)
-
-    if len(jobs) == 1:
-        msg = jobs[0]
-        job = json.loads(msg.body)
-
-        # The default Visibilty Timeout for a queue is 30 seconds. If we don't delete the
-        # message or update its Visibility Timeout in that time SQS will automatically
-        # requeue the message. This could lead to multiple workers working on
-        # the same job!
-        #
-        # While we could set the Visibility Timeout to some higher value (less than the 12 hour
-        # maximum), the approach taken here is to delete the message once it has been received.
-        # This means if processing fails for whatever reason we won't get the job automatically
-        # requeued. But this is a good thing since Whisper processing on a GPU is expensive.
-        # Instead exceptions will be caught and an error message will be sent to the Done queue.
-        #
-        # See: https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-visibility-timeout.html
-        msg.delete()
-
-        return job
-    elif len(jobs) == 0:
-        return None
-    else:
-        # this should never happen
-        raise SpeechToTextException(f"expected 1 job from queue but got {len(jobs)}")
```

Comment:
The previous polling SQS for work to do goes away since AWS Batch will run the Docker container with the `--job` parameter set to the JSON for the job, that was passed to Batch from common-accessioning.